isce2_topsapp/localize_burst.py

```diff
@@ -0,0 +1,315 @@
+        self.safe_name = Path(self.safe_url).with_suffix('.SAFE').name
+
+        image_numbers = [int(x.attrib['source_filename'].split('-')[-1][2]) for x in metadata]
+        products = [x.tag for x in metadata]
+        combos = list(zip(image_numbers, products))
+
+        files = {'product': 'annotation', 'calibration': 'calibration', 'noise': 'noise'}
+        for name in files:
+            elem = metadata[combos.index((self.image_number, name))]
+            content = copy.deepcopy(elem.find('content'))
+            content.tag = 'product'
+            setattr(self, files[name], content)
+            setattr(self, f'{files[name]}_name', elem.attrib['source_filename'])
+
+        self.manifest = manifest
+        self.manifest_name = 'manifest.safe'
+
+        file_paths = [x.attrib['href'] for x in manifest.findall('.//fileLocation')]
+        pattern = f'^./measurement/s1.*{self.image_number}.tiff$'
+        self.measurement_name = [Path(x).name for x in file_paths if re.search(pattern, x)][0]
+
+        self.gcp_df = self.create_gcp_df()
+        self.footprint = self.create_geometry(self.gcp_df)[0]
+        self.swath = int(self.annotation.findtext('.//{*}adsHeader/swath')[2])
+        self.polarisation = self.annotation.findtext('.//{*}adsHeader/polarisation')
+        self.orbit_direction = self.manifest.findtext('.//{*}pass').lower()
+
+    @staticmethod
+    def reformat_gcp(point):
+        attribs = ['line', 'pixel', 'latitude', 'longitude', 'height']
+        values = {}
+        for attrib in attribs:
+            values[attrib] = float(point.find(attrib).text)
+        return values
+
+    def create_gcp_df(self):
+        points = self.annotation.findall('.//{*}geolocationGridPoint')
+        gcp_df = pd.DataFrame([self.reformat_gcp(x) for x in points])
+        gcp_df = gcp_df.sort_values(['line', 'pixel']).reset_index(drop=True)
+        return gcp_df
+
+    def create_geometry(self, gcp_df):
+        burst_index = self.burst_number - 1
+        lines = int(self.annotation.findtext('.//{*}linesPerBurst'))
+        first_line = gcp_df.loc[gcp_df['line'] == burst_index * lines, ['longitude', 'latitude']]
+        second_line = gcp_df.loc[gcp_df['line'] == (burst_index + 1) * lines, ['longitude', 'latitude']]
+        x1 = first_line['longitude'].tolist()
+        y1 = first_line['latitude'].tolist()
+        x2 = second_line['longitude'].tolist()
+        y2 = second_line['latitude'].tolist()
+        x2.reverse()
+        y2.reverse()
+        x = x1 + x2
+        y = y1 + y2
+        footprint = geometry.Polygon(zip(x, y))
+        centroid = tuple([x[0] for x in footprint.centroid.xy])
+        return footprint, footprint.bounds, centroid
+
+
+def create_burst_request(burst_params: BurstParams, content: str) -> dict:
+    urls = {
+        'metadata': 'https://g6rmelgj3m.execute-api.us-west-2.amazonaws.com/metadata',
+        'geotiff': 'https://g6rmelgj3m.execute-api.us-west-2.amazonaws.com/geotiff',
+    }
```

Comment:
I think I'd hardcode this at the top of the file in all caps instead of inside this function to make it clear it's constant/magic string.

Reply:
Agreed. Will move.